modules/generator/instance.go

```diff
@@ -189,6 +191,13 @@ func (i *instance) updateProcessors() error {
 		return err
 	}
 
+	ingestionSlackInt := i.overrides.MetricsGeneratorIngestionSlack(i.instanceID).Nanoseconds()
+	if ingestionSlackInt == 0 {
+		ingestionSlackInt = i.cfg.MetricsIngestionSlack.Nanoseconds()
+	}
+
+	atomic.StoreInt64(&i.ingestionSlackOverride, ingestionSlackInt)
```

Comment:
use an atomic package over the primitives. it helps prevent making mistakes later. we mostly use: https://github.com/uber-go/atomic